src/uct/tcp/sockcm/sockcm_iface.c

```diff
@@ -68,7 +81,24 @@ static ucs_status_t uct_sockcm_iface_reject(uct_iface_h tl_iface,
 static ucs_status_t uct_sockcm_ep_flush(uct_ep_h tl_ep, unsigned flags,
                                         uct_completion_t *comp)
 {
-    return UCS_ERR_NOT_IMPLEMENTED;
+    uct_sockcm_ep_t    *ep = ucs_derived_of(tl_ep, uct_sockcm_ep_t);
+    ucs_status_t       status;
+    uct_sockcm_ep_op_t *op;
+
+    pthread_mutex_lock(&ep->ops_mutex);
+    status = ep->status;
+    if ((status == UCS_INPROGRESS) && (comp != NULL)) {
+        op = ucs_malloc(sizeof(*op), "uct_sockcm_ep_flush op");
```

Comment:
where is it called and freed?

Reply:
@brminich It's done here - https://github.com/openucx/ucx/blob/master/src/uct/tcp/sockcm/sockcm_ep.c#L39

and `uct_sockcm_ep_invoke_completions` is called in the event handler that's part of the follow-up PR here - 
https://github.com/openucx/ucx/pull/3570/files#diff-aca878e51b7d5e6eaa75cd20a1375eadR209

Reply:
:+1: